dist/js/page-specific/branchEmailSwitcher.js

```diff
@@ -6,64 +6,67 @@
 // Not using constants since I can only load one file on the form.
 
 var bcpl = window.bcpl || {};
+
 bcpl.pageSpecific = bcpl.pageSpecific || {};
 
-(function ($) {
-	var jqueryTag = document.createElement('script');
-	if (!$) {
-		jqueryTag.src = '/sebin/z/y/jquery.min.js';
-		document.querySelector('head').appendChild(jqueryTag);
-	}
-
-	var jqueryLoadInterval = setInterval(function () {
-		if (jQuery) {
-			$(function () {
-				bcpl.branchEmailSwitcher.init($);
-			});
-			clearInterval(jqueryLoadInterval);
-		}
-	}, 100);
-})(jQuery);
-
-bcpl.branchEmailSwitcher = function ($) {
-	var branchData = [];
-	var formResultMailFieldSelector = '#_seResultMail';
-
-	var findBranchEmail = function findBranchEmail(searchTerm) {
-		var foundEmail = branchData.find(function (branchEmailItem) {
-			if (branchEmailItem && branchEmailItem.myLibrarianEmail && typeof branchEmailItem.myLibrarianEmail === 'string') {
-				return branchEmailItem.name.toLowerCase() === searchTerm.toLowerCase();
-			}
-
-			return false;
-		});
-
-		return foundEmail || '';
-	};
-
-	var branchChangeHandler = function branchChangeHandler(changeEvent) {
-		var selectedBranch = changeEvent.target;
-		var branchSelectionValue = selectedBranch.value;
-		var branchEmailItem = findBranchEmail(branchSelectionValue);
-
-		$(selectedBranch).closest('form').find(formResultMailFieldSelector).attr('value', branchEmailItem.myLibrarianEmail);
-	};
-
-	var branchDataSuccessHandler = function branchDataSuccessHandler(branchJson) {
-		branchData = branchJson;
-	};
-
-	var branchDataErrorHandler = function branchDataErrorHandler(error) {
-		console.error(error);
-	};
-
-	var init = function init($) {
-		$.ajax('/sebin/q/r/branch-amenities.json').then(branchDataSuccessHandler, branchDataErrorHandler);
-
-		$(document).on('change', '#whichBranch', branchChangeHandler);
-	};
-
-	return {
-		init: init
-	};
-}(jQuery);
\ No newline at end of file
+window.onload = function () {
```

Comment:
Small problem, this will overwrite any other onload functions. If you use `$(window).on('load', fn)` or ` window.addEventListener("load", fn)` you'll be in better shape.

Reply:
good point, thanks will update